### PR TITLE
Support for Dio v4

### PR DESCRIPTION
**Pull request by ArneSaknussemm89**
## Description of the change
Updating the interceptor to work for Dio >= 4.0

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- 
### Development
- [X] Lint rules pass locally
- [X] The code changed/added as part of this pull request has been covered with tests
- 
### Code review 
- [ ] This pull request has a descriptive title and information useful to a reviewer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: lib/instabug_dio_interceptor.dart [hunk 1]
```diff
@@ -6,44 +6,51 @@ class InstabugDioInterceptor extends Interceptor {
...
-    data.responseBody = response.data;
-    data.responseHeaders = responseHeaders;
-    return data;
+    response.headers.forEach((String name, dynamic value) => responseHeaders[name] = value);
+    return data.copyWith(
+      endTime: DateTime.now(),
+      duration: data.endTime != null ? data.endTime!.millisecondsSinceEpoch - data.startTime.millisecondsSinceEpoch : 0,
+      url: response.requestOptions.uri.toString(),
+      method: response.requestOptions.method,
+      requestBody: response.requestOptions.data,
```
Comment (MahmoudMehisen) [inline lib/instabug_dio_interceptor.dart]: add toString() to request body data to work correctly with the native side

File: lib/instabug_dio_interceptor.dart [hunk 2]
```diff
@@ -6,44 +6,51 @@ class InstabugDioInterceptor extends Interceptor {
...
+      requestHeaders: response.requestOptions.headers,
+      contentType: response.requestOptions.contentType,
+      status: response.statusCode,
+      responseBody: response.data,
```
Comment (MahmoudMehisen) [inline lib/instabug_dio_interceptor.dart]: add toString() to response body data to work correctly with the native side

File: lib/instabug_dio_interceptor.dart (hunk 1 above)
Comment (ArneSaknussemm89) [reply lib/instabug_dio_interceptor.dart]: Done!

File: lib/instabug_dio_interceptor.dart (hunk 2 above)
Comment (ArneSaknussemm89) [reply lib/instabug_dio_interceptor.dart]: Done!

**Pull request merged**